contributingGuides/TS_STYLE.md

```diff
@@ -0,0 +1,515 @@
+  - [1.8 Optional chaining and nullish coalescing](#ts-nullish-coalescing)
+  - [1.9 Type Inference](#type-inference)
+  - [1.10 JSDoc](#jsdoc)
+  - [1.11 `propTypes` and `defaultProps`](#proptypes-and-defaultprops)
+  - [1.12 Utility Types](#utility-types)
+  - [1.13 `object` Type](#object-type)
+  - [1.14 Export Prop Types](#export-prop-types)
+  - [1.15 File Organization](#file-organization)
+  - [1.16 Reusable Types](#reusable-types)
+  - [1.17 `.tsx`](#tsx)
+  - [1.18 No inline prop types](#no-inline-prop-types)
+- [Communication Items](#items)
+- [Migration Guidelines](#migration-guidelines)
+
+## Other Expensify Resources on TypeScript
+
+- [Expensify TypeScript React Native CheatSheet](./TS_CHEATSHEET.md)
+- [Expensify TypeScript PropTypes Conversion Table](./PROPTYPES_CONVERSION_TABLE.md)
+
+## Learning Resources
+
+### Quickest way to learn TypeScript
+
+- Get up to speed quickly
+  - [TypeScript playground](https://www.typescriptlang.org/play?q=231#example)
+    - Go though all examples on the playground. Click on "Example" tab on the top
+- Handy Reference
+  - [TypeScript CheatSheet](https://www.typescriptlang.org/cheatsheets)
+    - [Type](https://www.typescriptlang.org/static/TypeScript%20Types-ae199d69aeecf7d4a2704a528d0fd3f9.png)
+    - [Control Flow Analysis](https://www.typescriptlang.org/static/TypeScript%20Control%20Flow%20Analysis-8a549253ad8470850b77c4c5c351d457.png)
+- TypeScript with React
+  - [React TypeScript CheatSheet](https://react-typescript-cheatsheet.netlify.app/)
+    - [List of built-in utility types](https://react-typescript-cheatsheet.netlify.app/docs/basic/troubleshooting/utilities)
+    - [HOC CheatSheet](https://react-typescript-cheatsheet.netlify.app/docs/hoc/)
+
+## Exception to Rules
+
+Most of the rules are enforced in ESLint or checked by TypeScript. If you think your particular situation warrants an exception, post the context in the `#expensify-open-source` Slack channel with your message prefixed with `TS EXCEPTION:`. Internal engineers will assess the case and suggest alternative or grants an exception. When an exception is granted, link the relevant Slack conversation in your pull request. Suppress ESLint or TypeScript warnings/errors with comments if necessary.
+
+This rule will apply until the migration is done. After the migration, exceptions are assessed and granted by PR reviewers.
+
+## General Rules
+
+Strive to type as strictly as possible.
+
+```ts
+type Foo = {
+  fetchingStatus: "loading" | "success" | "error"; // vs. fetchingStatus: string;
+  person: { name: string; age: number }; // vs. person: Record<string, unknown>;
+};
+```
+
+## Guidelines
+
+<a name="naming-conventions"></a><a name="1.1"></a>
+
+- [1.1](#naming-conventions) **Naming Conventions**: Follow naming conventions specified below
+
+  - Use PascalCase for type names. eslint: [`@typescript-eslint/naming-convention`](https://typescript-eslint.io/rules/naming-convention/)
+
+    ```ts
+    // bad
```

Comment:
```suggestion
    // BAD
```

@hayata-suenaga wdyt about uppercasing ALL bad / good words in the examples to make them more impactful and visible?

Reply:
nice idea gonna change 👍

Reply:
done